data_structures/binary_tree/binary_search_tree.py

```diff
@@ -69,12 +77,11 @@ def __insert(self, value):
                         parent_node = parent_node.right
             new_node.parent = parent_node
 
-    def insert(self, *values):
+    def insert(self, *values) -> None:
```

Comment:
Oh, this is not what I meant, sorry if I was not clear. When you're returning the type, it's not actually defined, so it will raise an exception when you run the code. To solve this problem, the linked PEP was proposed to postpone the evaluation of annotations like this.

The solution for now is to quote it:
```suggestion
    def insert(self, *values) -> "BinarySearchTree":
```

Update the return statement and doctest as well.

Reply:
I understand what you were saying, instead I realised that the fact that the `insert` statement was returning `self` was questionable, so I removed the return statement and made it return `None`

Reply:
> I am aware that the class is not defined at runtime, which is why I imported annotations to automatically encase the import in quotes

Yes, but this feature has been put on hold indefinitely, so we should not incorporate it. But not returning makes more sense, thanks!